src/gui/syncconflictsmodel.cpp

```diff
@@ -0,0 +1,355 @@
+    return mData;
+}
+
+bool SyncConflictsModel::allExistingsSelected() const
+{
+    return mAllExistingsSelected;
+}
+
+bool SyncConflictsModel::allConflictingSelected() const
+{
+    return mAllConflictingsSelected;
+}
+
+void SyncConflictsModel::setConflictActivities(ActivityList conflicts)
+{
+    if (mData == conflicts) {
+        return;
+    }
+
+    beginResetModel();
+
+    mData = conflicts;
+    emit conflictActivitiesChanged();
+
+    updateConflictsData();
+
+    endResetModel();
+}
+
+void SyncConflictsModel::selectAllExisting(bool selected)
+{
+    for (auto &singleConflict : mConflictData) {
+        singleConflict.mExistingSelected = selected;
+    }
+
+    Q_EMIT dataChanged(index(0), index(rowCount() - 1), {static_cast<int>(SyncConflictRoles::ExistingSelected)});
+
+    if (selected && !mAllExistingsSelected) {
+        mAllExistingsSelected = true;
+        Q_EMIT allExistingsSelectedChanged();
+    } else if (!selected && mAllExistingsSelected) {
+        mAllExistingsSelected = false;
+        Q_EMIT allExistingsSelectedChanged();
+    }
+}
+
+void SyncConflictsModel::selectAllConflicting(bool selected)
+{
+    for (auto &singleConflict : mConflictData) {
+        singleConflict.mConflictSelected = selected;
+    }
+
+    Q_EMIT dataChanged(index(0), index(rowCount() - 1), {static_cast<int>(SyncConflictRoles::ConflictSelected)});
+
+    if (selected && !mAllConflictingsSelected) {
+        mAllConflictingsSelected = true;
+        Q_EMIT allConflictingSelectedChanged();
+    } else if (!selected && mAllConflictingsSelected) {
+        mAllConflictingsSelected = false;
+        Q_EMIT allConflictingSelectedChanged();
+    }
+}
+
+void SyncConflictsModel::applyResolution()
+{
+    for(const auto &syncConflict : qAsConst(mConflictData)) {
+        if (syncConflict.isValid()) {
+            qCInfo(lcSyncConflictsModel) << syncConflict.mExistingFilePath << syncConflict.mConflictingFilePath << syncConflict.solution();
+            ConflictSolver solver;
+            solver.setLocalVersionFilename(syncConflict.mConflictingFilePath);
+            solver.setRemoteVersionFilename(syncConflict.mExistingFilePath);
+            solver.exec(syncConflict.solution());
+        }
+    }
+}
+
+void SyncConflictsModel::updateConflictsData()
+{
+    mConflictData.clear();
+    mConflictData.reserve(mData.size());
+
+    for (const auto &oneConflict : qAsConst(mData)) {
+        if (!FolderMan::instance()) {
+            qCWarning(lcSyncConflictsModel) << "no FolderMan instance";
+            mConflictData.push_back({});
+            continue;
+        }
+        const auto folder = FolderMan::instance()->folder(oneConflict._folder);
+        if (!folder) {
+            qCWarning(lcSyncConflictsModel) << "no Folder instance for" << oneConflict._folder;
+            mConflictData.push_back({});
+            continue;
+        }
+
+        const auto conflictedRelativePath = oneConflict._file;
+        const auto dbRecord = folder->journalDb();
```

Comment:
@mgallien `dbRecord` name is confusing, should be something like `jorunalDb`, or `localDb`

Reply:
done